backends-common/elasticsearch-v7/src/main/java/org/apache/james/backends/es/v7/IndexCreationFactory.java

```diff
@@ -94,38 +93,34 @@ public String getPayload() {
 
             @FunctionalInterface
             public interface RequireIndexName {
-                RequireAlias indexName(IndexName indexName);
-            }
-
-            @FunctionalInterface
-            public interface RequireAlias {
-                FinalStage addAlias(Collection<AliasName> aliases);
-
-                default FinalStage addAlias(AliasName... aliases) {
-                    return addAlias(ImmutableList.copyOf(aliases));
-                }
+                FinalStage indexName(IndexName indexName);
             }
 
             public static class FinalStage {
                 private final int nbShards;
                 private final int nbReplica;
                 private final int waitForActiveShards;
                 private final IndexName indexName;
-                private final ImmutableList<AliasName> aliases;
+                private final ImmutableList.Builder<AliasName> aliases;
                 private Optional<IndexCreationCustomElement> customAnalyzers;
                 private Optional<IndexCreationCustomElement> customTokenizers;
 
-                FinalStage(int nbShards, int nbReplica, int waitForActiveShards,
-                           IndexName indexName, Collection<AliasName> aliases) {
+                FinalStage(int nbShards, int nbReplica, int waitForActiveShards, IndexName indexName) {
                     this.nbShards = nbShards;
                     this.nbReplica = nbReplica;
                     this.waitForActiveShards = waitForActiveShards;
                     this.indexName = indexName;
-                    this.aliases = ImmutableList.copyOf(aliases);
+                    this.aliases = ImmutableList.builder();;
                     this.customAnalyzers = Optional.empty();
                     this.customTokenizers = Optional.empty();
                 }
 
+                public FinalStage addAlias(AliasName aliasName) {
```

Comment:
It does not stop you to keep this method and add an extra one like you had before `addAlias(AliasName... aliases)`

Good to have both :) Promiss after this I stop bothering you !